### ISSUE TITLE
General deep learning solver of agent optimization problem

### ISSUE THREAD

**Issue by sbenthall** [econ-ark/HARK#1129]
A nice-to-have feature based on the conversation today ...

A general solver to models defined using `HARK.frame` that uses a (deep) neural network for its solution/policy functions.

The idea is to use Tensorflow to simulation the model forward and use the resulting utility as a loss function, which is then reduced through deep learning training techniques.

Some work by Pablo for inspiration:

https://notes.quantecon.org/submission/5ddb3c926bad3800109084bf

**Comment by alanlujan91** [comment]
See also 

- https://github.com/sischei/DeepEquilibriumNets
- https://github.com/DeepSurrogate/OptionPricing

**Comment by sbenthall** [comment]
https://github.com/google-research/tensorflow_constrained_optimization/blob/master/README.md

**Comment by sbenthall** [comment]
https://github.com/dseconf/DSE2023/tree/main/06_Scheidegger

**Comment by akshayshanker** [comment]
Moving these ideas into long-run section of roadmap and closing.
